osutil/sys/syscall.go

```diff
@@ -0,0 +1,84 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package sys
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+// uid_t is an unsigned 32-bit integer in linux right now.
+// so syscall.Gete?[ug]id are wrong, and break in 32 bits
+// (see https://github.com/golang/go/issues/22739)
+func Getuid() uint32 {
+	return getid(syscall.SYS_GETUID)
+}
+
+func Geteuid() uint32 {
+	return getid(syscall.SYS_GETEUID)
```

Comment:
Shouldn't the *32 variants be used? 

Taking a look at https://github.com/golang/go/blob/master/src/syscall/zsyscall_linux_386.go#L1242 and https://github.com/golang/go/blob/master/src/syscall/zsyscall_linux_amd64.go the 386 variant is using SYS_GETUID32. Can't tell if this is something we need to do as well.

Reply:
that's a very good catch, which led me to write a test, which led me to https://github.com/golang/go/issues/22924 (and the current workaround). Thanks!